### PR TITLE
clippy: remove the unused field, elapsed, from SampleStats

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

some changes for bumping Rust version: https://github.com/anza-xyz/agave/pull/1309

Screenshot 2024-05-14 at 20 20 25

#### Summary of Changes

allow it as a stopgap 🫠

### CODE REVIEW

File: bench-tps/src/perf_utils.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@ use {
     },
 };
 
+#[allow(dead_code)]
```
Comment (steviez) [inline bench-tps/src/perf_utils.rs]: If it is only complaining about the `elapsed` field not being used, I would rather just remove the `elapsed` field.

I see that we set it here:
https://github.com/anza-xyz/agave/blob/6fb43389778b370fc434fac6e6fc6abd955af355/bench-tps/src/perf_utils.rs#L79-L82

But then doesn't look like we ever read `elapsed` in this function:
https://github.com/anza-xyz/agave/blob/6fb43389778b370fc434fac6e6fc6abd955af355/bench-tps/src/bench.rs#L1046-L1051

Comment (yihau) [reply bench-tps/src/perf_utils.rs]: fixed: [125bac2](https://github.com/anza-xyz/agave/pull/1346/commits/125bac23cdb2426fd416c408638c2e69cfd7dd84)

File: bench-tps/src/perf_utils.rs [hunk 2]
```diff
@@ -12,12 +12,11 @@ use {
...
```
Comment (steviez) [inline bench-tps/src/perf_utils.rs]: If we stripped out the unused field, we should no longer need this right ?

Comment (yihau) [reply bench-tps/src/perf_utils.rs]: ah... sorry. let me remove it now 😭

Comment (yihau) [reply bench-tps/src/perf_utils.rs]: fixed: [772b9be](https://github.com/anza-xyz/agave/pull/1346/commits/772b9be56c2061733767aca8c79312b13a6a4af5)

**Review by steviez — APPROVED** [review]
LGTM - was about to ask you to update the PR title but saw it update as I was typing this 😄

**Pull request merged**